pymc3/distributions/multivariate.py

```diff
@@ -191,6 +193,55 @@ def logp(self, x):
             T.all(x >= 0), T.all(x <= n), T.eq(T.sum(x), n),
             n >= 0)
 
+def posdef(AA):
+    try:
+        fct = np.linalg.cholesky(AA)
+        return 1
+    except np.linalg.LinAlgError:
+        return 0
+
+
+class PosDefMatrix(theano.Op):
+    """
+    Check if input is positive definite. Input should be a square matrix.
+
+    """
+
+    #Properties attribute
+    __props__ = ()
+
+
+    #Compulsory if itypes and otypes are not defined
+
+    def make_node(self, x):
+        x = theano.tensor.as_tensor_variable(x)
+        assert x.ndim == 2
+        o=T.TensorType(dtype='int8', broadcastable = [])()
+        return theano.Apply(self, [x], [o])
+
+    # Python implementation:
+    def perform(self, node, inputs, outputs):
+
+        (x,) = inputs
+        (z,) = outputs
+        try:
+            z[0] = np.array(posdef(x), dtype='int8')
+        except Exception:
+            print('Failed to check if positive definite', x)
+            raise
+
+    def infer_shape(self, node, shapes):
+        return [[]]
+
+    def grad(self, inp, grads):
+        x, = inp
+        return [x.zeros_like(theano.config.floatX)]
+
+    def __str__(self):
```

Comment:
It seem good overall. But personnaly, I don't like a \__str__ that don't return the class name. I think it make things more confusing when you search. So personally, I would remove that method.